source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,196 @@
+                                                                       n_compositional_fields);
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned() && cell->at_boundary())
+          {
+            unsigned int face_idx = numbers::invalid_unsigned_int;
+            bool at_upper_surface = false;
+            {
+              for (unsigned int f=0; f<GeometryInfo<3>::faces_per_cell; ++f)
+                {
+                  if (cell->at_boundary(f) && cell->face(f)->boundary_id() == top_boundary_id)
+                    {
+                      // If the cell is at the top boundary, assign face_idx.
+                      face_idx = f;
+                      at_upper_surface = true;
+                      break;
+                    }
+                }
+
+            }
+            fe_face_values.reinit(cell, face_idx);
+
+            if (at_upper_surface)
+              {
+                in.reinit(fe_face_values,
+                          cell,
+                          this->introspection(),
+                          this->get_solution());
+                //in.requested_properties = MaterialModel::MaterialProperties::viscosity;
+
+                this->get_material_model().fill_additional_material_model_inputs(in,
+                                                                                 this->get_solution(),
+                                                                                 fe_face_values,
+                                                                                 this->introspection());
+
+                this->get_material_model().evaluate(in, out);
+
+                for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    // Calculate the square root of the second moment invariant for the deviatoric strain rate tensor.
+                    //edot_ii = std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[q])))),
+                    //          min_strain_rate);
+
+                    const double local_second_invariant_of_strain_rate = (std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[q])))),min_strain_rate) * fe_face_values.JxW(q));
+                    const double local_surface_area = fe_face_values.JxW(q);
+
+                    // second invariant of strain rate over the whole surface
+                    local_second_invariant_of_strain_rate_integral += local_second_invariant_of_strain_rate;
+                    local_surface_area_integral += local_surface_area;
+                  }
+              }
+          }
+
+      double total_second_invariant_of_strain_rate (local_second_invariant_of_strain_rate_integral.size());
+      Utilities::MPI::sum (local_second_invariant_of_strain_rate_integral, this->get_mpi_communicator(), total_second_invariant_of_strain_rate);
```

Comment:
These two lines look incorrect: no need to initialize the double with a value. Also, doesn't sum() return the value (you are ignoring the return value)?

Reply:
So in essence this should be enough:
```suggestion
      const double total_second_invariant_of_strain_rate = Utilities::MPI::sum (local_second_invariant_of_strain_rate_integral, this->get_mpi_communicator(), total_second_invariant_of_strain_rate);
```

But also note that you use the name `local_second_invariant_of_strain_rate_integral` twice, once in line 55 (as a vector) once in line 114 (where it looks like you want to use it like a scalar double). What would you like to store in this variable?

Reply:
Sorry this should be a double. it should be "local_second_invariant_of_strain_rate * local_surface_area" integrated over the whole surface.